libs/topology/include/hpx/topology/topology.hpp

```diff
@@ -382,7 +378,12 @@ namespace hpx { namespace threads
 
         std::size_t num_of_pus_;
 
-        mutable hpx::util::spinlock topo_mtx;
+        using mutex_type = boost::detail::spinlock;
```

Comment:
Sorry @hkaiser, I misremembered how many places need to use boost spinlock directly. There shouldn't be many more though...

Reply:
In fact, the concurrency module will be next and we can make this depend on the concurrency module allowing us to keep `hpx::util::spinlock` here. @aurianer will take care of updating this.

Reply:
@aurianer let's change this back to `hpx::util::spinlock` now. (I think the timeout on the `numa_allocator` test comes from the Boost spinlock requiring a special initial value.)